### PR TITLE
chore: Deprecate `.findById` signature with `Optional` parameter

### PR DESCRIPTION
**Pull request by sharat87**
This is work on getting to remove the `.findById` signature with `Optional<>` arguments. This signature doesn't add any value, encourages confusing multi-signature service methods (check the diff here), and is causing unnecessary problems in `pg` branch with generating `*Cake` classes.

This PR doesn't get rid of this entirely, just one part. A follow-up PR will be opened after this is merged.

Nothing new, nothing fixed. Only a refactor.

No conflicts to EE, but needs extra changes, in [this PR](https://github.com/appsmithorg/appsmith-ee/pull/3714) to be merged for the build to pass.

**/test sanity**

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9549476417>
> Commit: c1e45f2fe50530b7ec8436d9722310537d125166
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9549476417&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`

## Summary by CodeRabbit

- **Refactor**
  - Simplified method signatures by removing the use of `Optional` for permission parameters across various services. Permissions are now passed directly.
  - Deprecated `findById` method that uses `Optional<AclPermission>` to improve code clarity and maintainability.

- **Chores**
  - Updated test cases to remove the use of `Optional.empty()` and replaced with `null` in method calls.
  - Removed unnecessary imports of `Optional` in multiple files.

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java [hunk 1]
```diff
@@ -431,7 +430,7 @@ public void updateDependencyMap_NotNullValue_shouldUpdateDependencyMap() {
                     dependencyMap.put("key3", List.of("val1", "val2"));
                     return newPageService
                             .updateDependencyMap(pageDTO.getId(), dependencyMap, null)
-                            .then(newPageService.findById(pageDTO.getId(), Optional.empty()));
+                            .then(newPageService.findById(pageDTO.getId(), null));
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java:433]: No. Passing `null` here has a distinctly separate meaning than the other signature that doesn't have this second argument. Passing `null` here explicitly states that we don't want permission checks to be verified for this query.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java [hunk 2]
```diff
@@ -468,7 +467,7 @@
                     return newPageService
                             .updateDependencyMap(pageDTO.getId(), dependencyMap, null)
                             .flatMap(page -> applicationPageService.publish(application.getId(), null, false))
-                            .then(newPageService.findById(pageDTO.getId(), Optional.empty()));
+                            .then(newPageService.findById(pageDTO.getId(), null));
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java:470]: Same as above as well.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java [hunk 3]
```diff
@@ -504,7 +503,7 @@
                 })
                 .flatMap(pageDTO -> newPageService
                         .updateDependencyMap(pageDTO.getId(), null, null)
-                        .then(newPageService.findById(pageDTO.getId(), Optional.empty())));
+                        .then(newPageService.findById(pageDTO.getId(), null)));
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/NewPageServiceTest.java:506]: Same as above as well.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java [hunk 4]
```diff
@@ -579,7 +579,7 @@ public Mono<ActionDTO> updateUnpublishedAction(String id, ActionDTO action) {
                 action != null ? action.getId() : null,
                 id);
 
-        return updateUnpublishedActionWithoutAnalytics(id, action, Optional.of(actionPermission.getEditPermission()))
+        return updateUnpublishedActionWithoutAnalytics(id, action, actionPermission.getEditPermission())
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:582]: Will be done in follow-up PRs. The scope of this PR is intentionally limited to be a small PR and be easy to review.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java [hunk 5]
```diff
@@ -14,6 +14,7 @@ public interface AppsmithRepository<T extends BaseDomain> {
 
     Mono<T> findById(String id, AclPermission permission);
 
+    @Deprecated(forRemoval = true)
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/AppsmithRepository.java:17]: Will be done in follow-up PRs. The scope of this PR is intentionally limited to be a small PR and be easy to review.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java [hunk 6]
```diff
@@ -343,28 +342,18 @@ public Mono<ActionCollectionDTO> update(String id, ActionCollectionDTO actionCol
...
     }
 
     @Override
     public Mono<ActionCollectionDTO> deleteUnpublishedActionCollection(String id) {
-        return deleteUnpublishedActionCollectionEx(
-                id,
-                Optional.of(actionPermission.getDeletePermission()),
-                Optional.of(actionPermission.getDeletePermission()));
+        return deleteUnpublishedActionCollection(
+                id, actionPermission.getDeletePermission(), actionPermission.getDeletePermission());
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java:351]: Will be done in follow-up PRs. The scope of this PR is intentionally limited to be a small PR and be easy to review.

**Review by trishaanand — APPROVED** [review]

**Pull request merged**